sw/device/tests/alert_handler_lpg_sleep_mode_pings.c

```diff
@@ -0,0 +1,425 @@
+  CHECK(irq_id == kDifPwrmgrIrqWakeup, "IRQ ID: %d is incorrect", irq_id);
+}
+
+bool test_main(void) {
+  // Enable global and external IRQ at Ibex.
+  irq_global_ctrl(true);
+  irq_external_ctrl(true);
+
+  init_peripherals();
+
+  // Enable all the AON interrupts used in this test.
+  rv_plic_testutils_irq_range_enable(&plic, kTopEarlgreyPlicTargetIbex0,
+                                     kTopEarlgreyPlicIrqIdPwrmgrAonWakeup,
+                                     kTopEarlgreyPlicIrqIdPwrmgrAonWakeup);
+
+  // Enable pwrmgr interrupt
+  CHECK_DIF_OK(dif_pwrmgr_irq_set_enabled(&pwrmgr, 0, kDifToggleEnabled));
+
+  // To keep the test results
+  bool is_cause;
+  bool is_locked;
+  // To keep the random number of iterations
+  uint32_t rnd_num_iterations;
+  // To keep track of the test steps
+  size_t test_step_cnt;
+
+  // Need a NVM counter to keep the test-step info
+  // after waking up from the deep sleep mode.
+  // Enable flash access
+  flash_ctrl_testutils_default_region_access(&flash_ctrl,
+                                             /*rd_en*/ true,
+                                             /*prog_en*/ true,
+                                             /*erase_en*/ true,
+                                             /*scramble_en*/ false,
+                                             /*ecc_en*/ false,
+                                             /*he_en*/ false);
+
+  test_step_cnt = flash_ctrl_testutils_counter_get(kCounterTestSteps);
+  // Total number of iterations for each test phase
+  rnd_num_iterations = flash_ctrl_testutils_counter_get(kCounterTestPhase);
+
+  // We don't expect test_step_cnt to 256 (flash mem is filled with all zeros)
+  // for this test. If it is 256, we just initialize the counters/NVM-fields
+  // again.
+  if (test_step_cnt == 256) {
+    flash_ctrl_testutils_counter_init_zero(&flash_ctrl, kCounterTestSteps);
+    flash_ctrl_testutils_counter_init_zero(&flash_ctrl, kCounterTestPhase);
+    test_step_cnt = flash_ctrl_testutils_counter_get(kCounterTestSteps);
+    rnd_num_iterations = flash_ctrl_testutils_counter_get(kCounterTestPhase);
+  }
+
+  // If this is the first iteration,
+  // set num_iterations to a random value between 8 and 32
+  if (rnd_num_iterations == 0) {
+    // 8 <= num_iters <= 32
+    rnd_num_iterations = rand_testutils_gen32_range(8, 32);
+    flash_ctrl_testutils_counter_set_at_least(&flash_ctrl, kCounterTestPhase,
+                                              rnd_num_iterations);
+    // LOG_INFO("phase1-num-iters = %d", rnd_num_iterations);
+  }
+
+  // Test phase #1: ping-timeout = 256
+  // No ping_timeout_fail alert shall be fired
+  // To ensure that the ping mechanism won't send spurious failure.
+  while (test_step_cnt < rnd_num_iterations) {
+    // Configure and lock the alerts
+    CHECK_DIF_OK(
+        dif_alert_handler_is_ping_timer_locked(&alert_handler, &is_locked));
+    if (!is_locked) {
+      alert_handler_config(/*ping_timeout=*/256);
+    }
```

Comment:
can we add a check here to make sure that the alert handler remains locked when coming back from light sleep?
otherwise this would just silently reconfigure the alert handler, and the test would pass.